src/main/java/com/google/cloud/tools/login/GoogleLoginState.java

```diff
@@ -380,11 +381,22 @@ public void initialize(HttpRequest httpRequest) throws IOException {
   private void persistCredentials() {
     Preconditions.checkState(isLoggedIn());
 
+    List<Exception> errors = new ArrayList<>();
     for (Account account : accountRoster.getAccounts()) {
       OAuthData oAuthData = new OAuthData(account.getAccessToken(), account.getRefreshToken(),
           account.getEmail(), account.getName(), account.getAvatarUrl(),
           oAuthScopes, account.getAccessTokenExpiryTime());
-      authDataStore.saveOAuthData(oAuthData);
+      try {
+        authDataStore.saveOAuthData(oAuthData);
```

Comment:
I think we need to bubble BackingStoreExceptions from here that we're currently catching, and then catch the more specific exceptions: BackingStoreException and SecurityException. Either that, or chain both of those into an IOException.